### ISSUE TITLE
Config to add strings

### ISSUE THREAD

**Issue by bytehala** [gkiely/eslint-plugin-jsx-no-leaked-render#2]
Hi, I found a reddit thread where you mention the possibility of adding strings as errors:

https://www.reddit.com/r/reactjs/comments/x4iipz/comment/imx220n/

```
"jsx-no-leaked-values/jsx-no-leaked-values": [
  "error", 
  {
    "string": true,
  }
]
```

This would be really helpful in React Native because rendering strings in RN will cause a crash:
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-leaked-render.md#rule-details

```
const Example = () => {
  return (
    <>
      {0 && <Something />}
      {/* React: renders undesired 0 */}
      {/* React Native: crashes 💥 */}

      {NaN && <Something />}
      {/* React: renders undesired NaN */}
      {/* React Native: crashes 💥 */}

      {'' && <Something />}
      {/* React: renders nothing */}
      {/* React Native, with React < 18: crashes 💥 */}
    </>
  )
}
```

**Comment by gkiely** [comment]
@bytehala 
Could you confirm the following:

It should report an error if:
1. It is a known constant string literal value of `''`
2. If it is a type of string (as we cannot know if it's empty or not)

So it would error for these 2 examples:
```ts
const Example = () => {
  const s = '';
  return (
    <>
      {s && <Something />}  // Known value of '', report error
    </>
  );
};

const Example = () => {
  let s = ''; 
  return (
    <>
      {s && <Something />} // Type of string, we cannot determine if it's empty so report error
    </>
  );
};
```

It would not error for a constant truthy string literal:
```ts
const Example = () => {
  const s = 'hello'; 
  return (
    <>
      {s && <Something />} // Known value of 'hello', do not report error
    </>
  );
};
```

**Comment by iway1** [comment]
@gkiely your approach would be really good. TBH checking for constant truth literal isn't even 100% necessary but would be nice to have
